File: data-tool/src/main/java/com/nilecochen/fitnessapp/datatool/entities/Entry.java
package com.nilecochen.fitnessapp.datatool.entities;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import java.util.HashSet;
import java.util.Set;

@Entity
@Table(name = "entries")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Entry {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "workout_exercise_id", nullable = false)
    private WorkoutExercise workoutExercise;

    @Column(name = "entry_index", nullable = false)
    private Long entryIndex;

    // Relationships
    @OneToMany(mappedBy = "entry", cascade = CascadeType.ALL, orphanRemoval = true)
    private Set<EntryMetric> entryMetrics = new HashSet<>();
}

File: admin-dash/Admin-Dash/Models/Entry.cs
using System;
using System.Collections.Generic;

namespace Admin_Dash.Models;

public partial class Entry
{
    public long Id { get; set; }

    public long WorkoutExerciseId { get; set; }

    public long EntryIndex { get; set; }

    public virtual ICollection<EntryMetric> EntryMetrics { get; set; } = new List<EntryMetric>();

    public virtual WorkoutExercise WorkoutExercise { get; set; } = null!;
}
